include/deal.II/lac/sparsity_pattern.h

```diff
@@ -1430,8 +1430,13 @@ SparsityPattern::load (Archive &ar, const unsigned int)
 
   ar &max_dim &rows &cols &max_vec_len &max_row_length &compressed &store_diagonal_first_in_row;
 
-  rowstart = new std::size_t [max_dim + 1];
-  colnums = new size_type [max_vec_len];
+  if (rowstart != 0)
+    delete[] rowstart;
+  rowstart = new std::size_t[max_dim + 1];
+
+  if (colnums != 0)
+    delete[] colnums;
+  colnums = new size_type[max_vec_len];
```

Comment:
While we use this check and delete idiom all over the place (and it is not wrong) it is worth noting that it is always okay to `delete` zero:
http://www.stroustrup.com/bs_faq2.html#delete-zero